game/src/Systems/AiMovementSystem.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <vengine.h>
+#include "../Components/AiMovement.h"
+
+#include <chrono>
+
+class AiMovementSystem : public System {
+private:
+	SceneHandler* sceneHandler;
+    Scene* scene;
+public:
+    AiMovementSystem(SceneHandler* sceneHandler): sceneHandler(sceneHandler)
+    {
+        scene = sceneHandler->getScene();
+        auto view = scene->getSceneReg().view<AiMovement>();
+        auto movementLambda = [&](AiMovement& movement) {
+            movement.maxSpeed = 50.f;
+            movement.speedIncrease = 10.f;
+            movement.slowDown = 180.f;
+            movement.currentSpeed = glm::vec2(0.f);
+            movement.turnSpeed = 200.f;
+        };
+        view.each(movementLambda);
+    }
+
+    bool update(entt::registry& reg, float dt) final
+    {
+        auto view = reg.view<Transform, AiMovement>();
+        auto moveLambda = [&](Transform& transform, AiMovement& movement) {
+            move(movement, transform, dt);
+            rotate(movement, transform, dt);
+        };
+        view.each(moveLambda);
+        return false;
+    }
+
+    void move(AiMovement& movement, Transform& transform, float dt)
+    {
+        Entity mainCamID = scene->getMainCameraID();
+        if (scene->hasComponents<Script>(mainCamID))
+        {
+            int playerID;
```

Comment:
Save as member variable and only call getScriptComponentValue once. Perhaps would be done if playerID == -1 or similar.